scripts/Inelastic/IndirectReductionCommon.py

```diff
@@ -1009,6 +1058,54 @@ def get_multi_frame_rebin(workspace_name, rebin_string):
 # -------------------------------------------------------------------------------
 
 
+def _get_x_range_when_bins_vary(workspace: MatrixWorkspace, grouping_workspace: GroupingWorkspace) -> Tuple[float]:
```

Comment:
I think the MaxMin algorithm gets the max and min Y value rather than the max and min X, but I agree that this functionality might be useful elsewhere

I have noticed this function on MatrixWorkpace
https://github.com/mantidproject/mantid/blob/36cb74141910af5a5146bc5011c654f6aca53668/Framework/API/src/MatrixWorkspace.cpp#L684

I'll see if I can add the ability to exclude masked detectors to this function, and then call it from here

Reply:
Right, I though it was also checking X ranges.